### PR TITLE
Add/update scripts for converting ACE -> HDF5, update documentation

### PR DESCRIPTION
**Pull request by paulromano**
This pull request makes the following changes:

- Adds two new scripts, `convert_mcnp_70.py` and `convert_mcnp_71.py`, that automatically convert ENDF/B-VII.0 and ENDF/B-VII.1 ACE libraries from MCNP5/6.
- Updates the `get_jeff_data.py` script to smartly convert data (avoid repeated calls to `from_hdf5()` which saves *a lot* of time)
- Updates the user's guide with information on converting ACE cross section libraries (per request of @bforget)
- Refactors `ThermalScattering.add_temperature_from_ace()` in the same way that `IncidentNeutron.add_temperature_from_ace()` was refactored earlier.

### CODE REVIEW

**Comment by paulromano** [comment]
Just added a commit fixing a bug when trying to calculate microscopic MGXS using nuclide densities calculated from a stochastic volume calc. Thanks to Nicolas Stauff at ANL/NE for reporting.

File: data/convert_mcnp_71.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                    help='Directory to create new library in')
+parser.add_argument('-f', '--fission_energy_release',
+                    help='HDF5 file containing fission energy release data')
+parser.add_argument('mcnpdata', help='Directory containing endf71x and ENDF71SaB')
+args = parser.parse_args()
+assert os.path.isdir(args.mcnpdata)
+
+# Get a list of all ACE files
+endf71x = glob.glob(os.path.join(args.mcnpdata, 'endf71x', '*', '*.71?nc'))
+endf71sab = glob.glob(os.path.join(args.mcnpdata, 'ENDF71SaB' , 'lwtr*.2?t'))
```
Comment (smharper) [inline data/convert_mcnp_71.py]: This is only capturing the light water S(a, b) tables.

File: data/convert_mcnp_71.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    print('Converting: ' + filename)
+    if filename.endswith('t'):
+        data = openmc.data.ThermalScattering.from_ace(filename)
+    else:
+        data = openmc.data.IncidentNeutron.from_ace(filename, 'mcnp')
+
+        # Add fission energy release data, if available
+        if args.fission_energy_release is not None:
+            fer = openmc.data.FissionEnergyRelease.from_compact_hdf5(
+                args.fission_energy_release, neutron)
```
Comment (smharper) [inline data/convert_mcnp_71.py]: `neutron => data`

**Comment by smharper** [comment]
I'm currently getting this error with the 7.1 library:
```
 ERROR: Cross-section table c_H_in_H2O is not a S(a,b) table.
```

Haven't found the root cause of this error yet;  I'll let you know when I do.

**Comment by paulromano** [comment]
It was an error in a call to `DataLibrary.register_file()`. I've simplified how that method works to avoid mistakes like this in the future.

**Comment by paulromano** [comment]
Just added a little fix that allows temperatures to be assigned to heterogeneous cells (the temperature is propagated to contained cells that actually have materials assigned)

**Comment by smharper** [comment]
Looks good.  Thanks for the PR!

**Pull request merged**